File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        start = 0
        subStr = set()
        maxSize = 0
        for end in range(len(s)):
            currLet = s[end]
            if currLet in subStr:
                while s[start] != currLet:
                    subStr.remove(s[start])
                    start += 1
                start += 1
            else:
                subStr.add(currLet)
                if (len(subStr) > maxSize):
                    maxSize = len(subStr)
        return maxSize

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Map<Character, Integer> letCount = new HashMap<>();
        int left = 0;
        int maxLength = 0;
        for (int right = 0; right < s.length(); right++)    {
            char c = s.charAt(right);
            if (letCount.containsKey(c))   {
                int remCharInd = letCount.get(c);
                while (left <= remCharInd)  {
                    letCount.remove(s.charAt(left++));
                } 
            }
            letCount.put(c, right);
            maxLength = Math.max(right - left + 1, maxLength);
        }
        return maxLength;

    }
}